packages/base-zone/src/prepare-revocable.js

```diff
@@ -128,11 +138,36 @@ export const prepareRevocableKit = (
       stateShape: {
         underlying: M.opt(M.remotable('underlying')),
       },
+      receiveAmplifier: amp => {
+        amplifier = amp;
+      },
     },
   );
 
-  // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
-  // @ts-ignore parameter confusion
-  return makeRevocableKit;
+  /**
+   * @param {U} underlying
+   * @returns {U}
+   */
+  const makeRevocable = underlying =>
+    // @ts-expect-error some confusion about UU vs Guarded<U> I think
```

Comment:
Actually, it's because of `extraMethods`.  The type `U` doesn't have them, but the return type of this function does.  Better to remove the `@returns` annotation and let it be inferred.

That will get you part of the way.  More typing work is probably needed in order to propagate the underlying and extra methods without just thinking they're a homogeneous `Record<string|symbol, (...args: any[]) => any>`.

Reply:
@michaelfig 
I tried to do the joint set of typing changes you suggest, and got stuck. As you say, more typing work is needed.